infra/testApi.mjs

```diff
@@ -3,176 +3,129 @@ import { strict as assert } from 'node:assert';  // use strict equality everywhe
+  assertIdAndPath(sample_json);
+  checkFelids(sample_json, ['command', 'points', 'job', 'path']);
+  assert.equal(sample_json.points.length, SAMPLE_SIZE);
+  return sample_json;
 }
-const explain = await (await fetch(makeEndpoint("/api/explanations"), explainBody)).json()
-assertIdAndPath(explain)
-assert.equal(explain.explanation.length > 0, true, 'explanation should not be empty');
-const explainAsyncResult = await callAsyncAndWaitJSONResult("/api/start/explanations", explainBody)
-assertIdAndPath(explainAsyncResult)
-assert.equal(explainAsyncResult.explanation.length > 0, true, 'explanation should not be empty');
-
-// Analyze endpoint
-const errorsBody = {
-  method: 'POST', body: JSON.stringify({
-    formula: FPCoreFormula, sample: [[[
-      14.97651307489794
-    ], 0.12711304680349078]]
-  })
+
+const sample1 = await callSample(FPCoreFormula, true);
+const sample2 = await callSample(FPCoreFormula, false);
+assert.deepEqual(sample1.points[1], sample2.points[1]);
+
+// MARK: Explanations
+async function callExplanations(fpcore, p_context, async_huh) {
+  const explain_json = await getJSONFor('explanations', 'POST', JSON.stringify({
+    formula: fpcore, sample: p_context
+  }), async_huh);
+  assertIdAndPath(explain_json);
+  checkFelids(explain_json, ['command', 'explanation', 'job', 'path']);
+  return explain_json;
 }
-const errors = await (await fetch(makeEndpoint("/api/analyze"), errorsBody)).json()
-assertIdAndPath(errors)
-assert.deepEqual(errors.points, [[[14.97651307489794], "2.3"]])
-const analyzeAsyncResult = await callAsyncAndWaitJSONResult("/api/start/analyze", errorsBody)
-assertIdAndPath(analyzeAsyncResult)
-assert.deepEqual(analyzeAsyncResult.points, [[[14.97651307489794], "2.3"]])
-
-// Exacts endpoint
-const exactsBody = {
-  method: 'POST', body: JSON.stringify({
-    formula: FPCoreFormula2, sample: eval_sample
-  })
+
+// MARK: Exacts endpoint
+async function callExacts(fpcore, p_context, async_huh) {
+  const exacts_json = await getJSONFor('exacts', 'POST', JSON.stringify({
+    formula: fpcore, sample: p_context
+  }), async_huh);
+  assertIdAndPath(exacts_json);
+  checkFelids(exacts_json, ['command', 'points', 'job', 'path']);
+  return exacts_json;
 }
-const exacts = await (await fetch(makeEndpoint("/api/exacts"), exactsBody)).json()
-assertIdAndPath(exacts)
-assert.deepEqual(exacts.points, [[[1], -1.4142135623730951]])
-const exactsAsyncResult = await callAsyncAndWaitJSONResult("/api/start/exacts", exactsBody)
-assertIdAndPath(exactsAsyncResult)
-assert.deepEqual(exactsAsyncResult.points, [[[1], -1.4142135623730951]])
-
-// Calculate endpoint
-const calculateBody = {
-  method: 'POST', body: JSON.stringify({
-    formula: FPCoreFormula2, sample: eval_sample
-  })
+
+const exacts1 = await callExacts(FPCoreFormula2, eval_sample, true);
+const exacts2 = await callExacts(FPCoreFormula2, eval_sample, false);
```

Comment:
I think every test is repeated twice, once in async mode and once not. Maybe build that into `getJSONFor` instead of having to do it manually?